pkg/pbq/store/wal/stores.go

```diff
@@ -35,13 +36,17 @@ type walStores struct {
 	maxBatchSize int64
 	// syncDuration timeout to sync to store
 	syncDuration time.Duration
+	pipelineName string
+	vertexName   string
 }
 
-func NewWALStores(opts ...Option) store.StoreProvider {
+func NewWALStores(pipelineName string, vertexName string, opts ...Option) store.StoreProvider {
```

Comment:
Pass VertexInstance? That way we will get more info including the replica number.